server/get-info-from-googleapis.js

```diff
@@ -0,0 +1,92 @@
+const { google } = require('googleapis');
+const { spreadSheet } = require('./config');
+const _ = require('lodash');
+
+const TEAM_SHEET_ID = spreadSheet.id;
+const CREDS_FILE = spreadSheet.securityFile;
+const scopes = ['https://www.googleapis.com/auth/drive.readonly'];
+const TEAM_RANGE = spreadSheet.teamRange;
+const CAREERS_RANGE = spreadSheet.careersRange;
+
+const ranges = [TEAM_RANGE, CAREERS_RANGE];
+
+const TITLE_COLUMN_INDEX = 0;
+
+let dataStore = null;
+let updateTime = null;
+const timeToLive = 10 * 60 * 1000; // 10min
+
+const arrayOfArraysToCollection = (arr) => {
+  const properties = arr[TITLE_COLUMN_INDEX];
+  return arr.slice(TITLE_COLUMN_INDEX + 1)
+    .map(v => _.fromPairs(properties.map((property, index) => ([property, v[index]]))));
+};
+
+const isValidData = () => {
+  if (!dataStore || !updateTime) {
+    return false;
+  }
+
+  const diffTime = +new Date() - updateTime;
+  return diffTime < timeToLive;
+};
+
+const updateData = (data) => {
+  dataStore = data;
+  updateTime = +new Date();
+};
+
+const getSheetValues = (valueRanges, rangeName) =>
+  valueRanges.find(v => new RegExp(rangeName).test(v.range)).values;
+
+const getData = async () => {
+  if (isValidData()) {
+    return Promise.resolve(dataStore);
+  }
+
+  try {
+    const client = await google.auth.getClient({
+      keyFile: CREDS_FILE,
+      scopes,
+    });
+
+    const sheets = google.sheets('v4');
+
+    const table = await sheets.spreadsheets.values.batchGet({
+      auth: client,
+      spreadsheetId: TEAM_SHEET_ID,
+      ranges,
+    });
```

Comment:
@dmytropanontko , let's change an approach a bit:

1. Let's cache `table.data.valueRanges`, not extracted values
2. Move fetching that table to separate function and apply [_.memoize](https://lodash.com/docs/4.17.11#memoize) with a resolver to it
3. getTeam and getCareers should pass range as a parameter to the parent function and get actual team not an object which they need to manipulate